### PR TITLE
DDF-2173 upgrade dependencies to fix OWASP findings

### PR DESCRIPTION
**Pull request by ryeats**
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@tbatie @bcwaters @adimka
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@shaundmorris
#### How should this be tested?
run owasp and itests, start ddf up and try to ingest and search
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: dependency-check-maven-config.xml [hunk 1]
```diff
@@ -84,4 +128,103 @@
    ]]></notes>
         <cve>CVE-2014-0003</cve>
     </suppress>
+
+    <suppress>
+        <notes><![CDATA[
+        this CVE is generating a lot of false positives it should only include jackson-dataformat-xml jar
+         but it is catching all jackson dependencies unfortunately I can't fix it so if we start depending
+          on a vulnerable version of jackson-dataformat-xml this will sill suppress it.
```
Comment (shaundmorris) [inline dependency-check-maven-config.xml]: s/b `still`

File: distribution/ddf-common/src/main/resources/common-bin.xml [hunk 2]
```diff
@@ -51,6 +51,26 @@
                 <exclude>RELEASE-NOTES</exclude>
                 <exclude>karaf-manual*.html</exclude>
                 <exclude>karaf-manual*.pdf</exclude>
+
+                <!-- excluding jars we don't need that have OWASP vulnerabilities -->
+                <!--
+                Excluding this unused jar, as we don't use it (TO BE CONFIRMED)
+                and it's associated with CVE-2004-0009
+                -->
+                <exclude>**/bundles/org.apache.servicemix.bundles.not-yet-commons-ssl/**</exclude>
```
Comment (stustison) [inline distribution/ddf-common/src/main/resources/common-bin.xml]: We do use this jar and it is needed for our certificate UI to work correctly. It is embedded within that jar, however, so it shouldn't be getting deployed anyways.

Comment (stustison) [inline distribution/ddf-common/src/main/resources/common-bin.xml]: This is also required for the karaf shell to work correctly. I don't think we can exclude this.

Comment (ryeats) [inline distribution/ddf-common/src/main/resources/common-bin.xml]: Now that I look at the CVE it looks unrelated to this bundle OWASP is picking up anything with apache and ssl in the name. Ill remove this and add a suppression.

File: dependency-check-maven-config.xml [hunk 3]
```diff
@@ -1,7 +1,41 @@
...
+        <cve>CVE-2015-5344</cve>
+    </suppress>
+
+    <suppress>
+        <notes><![CDATA[
+            CVE-2014-1939 is an Android specific issue, and does not apply here
+            file name: google-http-client-1.22.0.jar]]></notes>
+        <cve>CVE-2014-1939</cve>
+    </suppress>
+    <!--
```
Comment (coyotesqrl) [inline dependency-check-maven-config.xml]: Why is this a comment and not a note?

File: dependency-check-maven-config.xml [hunk 4]
```diff
@@ -63,6 +99,14 @@
...
 
+    <suppress>
+        <notes><![CDATA[
+   file name: solr-*.jar
+   OWASP is getting confused by our version number being on a jar with solr in the name we are on solr 6.0+ which is not affected by this issue.
+   ]]></notes>
+        <cve>CVE-2012-6612</cve>
+    </suppress>
+
     <!--
```
Comment (coyotesqrl) [inline dependency-check-maven-config.xml]: Not part of your PR, but close enough to tag. Should be a note, not a comment.

File: dependency-check-maven-config.xml (hunk 1 above)
Comment (ryeats) [inline dependency-check-maven-config.xml]: fixed

Comment (coyotesqrl) [inline dependency-check-maven-config.xml]: Could you maybe specify the file hashes of the jars in question to mitigate the risk that we might someday add a vulnerable `jackson-dataformat-xml` dependency?

File: dependency-check-maven-config.xml (hunk 3 above)
Comment (ryeats) [inline dependency-check-maven-config.xml]: ill fix all of these

File: dependency-check-maven-config.xml (hunk 4 above)
Comment (ryeats) [inline dependency-check-maven-config.xml]: fixed

File: dependency-check-maven-config.xml [hunk 5]
```diff
@@ -84,4 +128,103 @@
...
+    <suppress>
+        <notes><![CDATA[
+   file name: commons-beanutils-1.8.3.jar
+   shiro-core has a dependency on this but it doesn’t expose commons-beanutils to user input so it wouldn't pose a risk like the struts library that is called out in the CVE
+   ]]></notes>
+        <sha1>686EF3410BCF4AB8CE7FD0B899E832AABA5FACF7</sha1>
+        <cve>CVE-2014-0114</cve>
+    </suppress>
+
+    <!--todo remove this to ascertain what issues remain -->
```
Comment (shaundmorris) [inline dependency-check-maven-config.xml]: Is this todo still valid?

File: dependency-check-maven-config.xml [hunk 6]
```diff
@@ -84,4 +128,103 @@
...
+        <notes><![CDATA[
+        false positive CVE is unrelated
+   file name: nagasena-0000.0002.0049.0.jar
+   ]]></notes>
+        <sha1>8821DF6D1BC22FBA8F608021F4F27698CF3FD91F</sha1>
+        <cve>CVE-2014-9389</cve>
+    </suppress>
+
+
+    <!-- these are the geowebcache vulnerabilities it is not installed by default and it is only experimental these security issues would need to be resolved before geowebcache can be installed in a production enviroment-->
```
Comment (shaundmorris) [inline dependency-check-maven-config.xml]: Should this comment actually be in a feature description or somewhere visible in the Admin UI?

File: dependency-check-maven-config.xml [hunk 7]
```diff
@@ -84,4 +128,103 @@
...
+   ]]></notes>
+        <cve>CVE-2016-3720</cve>
+    </suppress>
+
+    <suppress>
+        <notes><![CDATA[
+   file name: commons-beanutils-1.8.3.jar
+   shiro-core has a dependency on this but it doesn’t expose commons-beanutils to user input so it wouldn't pose a risk like the struts library that is called out in the CVE
+   ]]></notes>
+        <sha1>686EF3410BCF4AB8CE7FD0B899E832AABA5FACF7</sha1>
```
Comment (coyotesqrl) [inline dependency-check-maven-config.xml]: Is this the has of `shiro-core`? If so and we upgrade `shiro-core` then I assume this vulnerability will re-present itself, correct?

File: dependency-check-maven-config.xml (hunk 5 above)
Comment (ryeats) [inline dependency-check-maven-config.xml]: yeah ill write another ticket

**Comment by stustison** [comment]
We also need to update Spring to 3.2.14.RELEASE

File: dependency-check-maven-config.xml (hunk 7 above)
Comment (ryeats) [inline dependency-check-maven-config.xml]: correct

**Comment by stustison** [comment]
The platform pom also has the incorrect versions of the httpcore, httpclient, and httpmime jars in its dependency management section.

File: distribution/test/itests/test-itests-ddf/pom.xml [hunk 8]
```diff
@@ -96,13 +96,13 @@
         <dependency>
             <groupId>org.apache.httpcomponents</groupId>
             <artifactId>httpcore</artifactId>
-            <version>4.3.3</version>
+            <version>4.4.4</version>
```
Comment (coyotesqrl) [inline distribution/test/itests/test-itests-ddf/pom.xml]: Any reason we wouldn't go to 4.4.5?

File: distribution/test/itests/test-itests-ddf/pom.xml [hunk 9]
```diff
@@ -96,13 +96,13 @@
...
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.httpcomponents</groupId>
             <artifactId>httpclient</artifactId>
-            <version>4.3.6</version>
+            <version>4.5.1</version>
```
Comment (coyotesqrl) [inline distribution/test/itests/test-itests-ddf/pom.xml]: Any reason we wouldn't go to 4.5.2?

File: distribution/test/itests/test-itests-ddf/pom.xml (hunk 8 above)
Comment (ryeats) [inline distribution/test/itests/test-itests-ddf/pom.xml]: nope fixed.

File: distribution/test/itests/test-itests-ddf/pom.xml (hunk 9 above)
Comment (ryeats) [inline distribution/test/itests/test-itests-ddf/pom.xml]: nope fixed.

File: distribution/test/thirdparty/restito/pom.xml [hunk 10]
```diff
@@ -28,7 +28,7 @@
         <dependency>
             <groupId>com.xebialabs.restito</groupId>
             <artifactId>restito</artifactId>
-            <version>0.7</version>
+            <version>0.8.2</version>
```
Comment (coyotesqrl) [inline distribution/test/thirdparty/restito/pom.xml]: Maybe make this a property in the `ddf.test/test` pom or even use dependency management?

File: platform/solr/platform-solr-server-standalone/dependency-check-maven-config.xml [hunk 11]
```diff
@@ -1,16 +1,37 @@
...
+         but it is catching all jackson dependencies unfortunately I can't fix it so if we start depending
+          on a vulnerable version of jackson-dataformat-xml this will sill suppress it.
+   ]]></notes>
+        <cve>CVE-2016-3720</cve>
+    </suppress>
+
     <!--
-        CVE-2011-2730 only applies when using JSP Expression Language (EL) which is not used in DDF.
-     -->
+    CVE-2008-0660 is a stack based buffer overflow vulnerability related to ActiveX and
```
Comment (coyotesqrl) [inline platform/solr/platform-solr-server-standalone/dependency-check-maven-config.xml]: Same re: note not comment.

File: platform/solr/platform-solr-server-standalone/dependency-check-maven-config.xml [hunk 12]
```diff
@@ -1,16 +1,37 @@
...
+-->
     <suppress>
-        <cve>CVE-2011-2730</cve>
+        <cve>CVE-2008-0660</cve>
     </suppress>
 
     <!--
-        CVE-2014-0050: Applies to commons-fileupload-1.2.1, suppressing due to replacing jar when packaging war
-    -->
+    CVE-2011-2730 is related to a vulnerability in the VMware SpringSource Spring
```
Comment (coyotesqrl) [inline platform/solr/platform-solr-server-standalone/dependency-check-maven-config.xml]: Same re: note not comment.

File: distribution/test/thirdparty/restito/pom.xml (hunk 10 above)
Comment (ryeats) [inline distribution/test/thirdparty/restito/pom.xml]: fixed

File: platform/solr/platform-solr-server-standalone/dependency-check-maven-config.xml (hunk 11 above)
Comment (ryeats) [inline platform/solr/platform-solr-server-standalone/dependency-check-maven-config.xml]: fixed

File: platform/solr/platform-solr-server-standalone/dependency-check-maven-config.xml (hunk 12 above)
Comment (ryeats) [inline platform/solr/platform-solr-server-standalone/dependency-check-maven-config.xml]: fixed

File: catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SendEventTest.java [hunk 13]
```diff
@@ -234,6 +236,7 @@ public void testIsAvailableSubjectExperation() throws Exception {
     }
 
     @Test
+    @Ignore
```
Comment (shaundmorris) [inline catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SendEventTest.java]: Is there a comment we can put in as to why this is ignored? just thinking of when it gets picked back up or does that no make any sense?

**Comment by shaundmorris** [comment]
On at least httpcore, httpclient seems like dependency management would be beneficial. Updates to platform, catalog, and itests were needed. Just a thought, not required for this PR.

File: catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SendEventTest.java (hunk 13 above)
Comment (ryeats) [inline catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SendEventTest.java]: yeah ill write a ticket for it and comment on it. It was intermittently failing intermittently.

**Comment by shaundmorris** [comment]
I just deployed geotools-suite 8.4_2.jar into Codice Nexus https://github.com/codice/thirdparty/pull/7

**Comment by stustison** [comment]
👍

**Comment by shaundmorris** [comment]
👍

**Pull request merged**